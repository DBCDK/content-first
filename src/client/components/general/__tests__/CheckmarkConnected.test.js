import React from 'react';
import renderer from 'react-test-renderer';

import {CheckMarkConnected} from '../CheckmarkConnected.component';

describe('CheckMarkConnected', () => {
  it('renders initial component', () => {
    const tree = renderer
      .create(
        <CheckMarkConnected
          shortListState={{elements: []}}
          isLoggedIn={false}
          book={{book: {pid: '0123'}}}
          origin="Fra egen værkside"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders initial component with loggedIn user', () => {
    const tree = renderer
      .create(
        <CheckMarkConnected
          shortListState={{elements: []}}
          isLoggedIn={true}
          book={{book: {pid: '0123'}}}
          origin="Fra egen værkside"
          systemLists={[]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
