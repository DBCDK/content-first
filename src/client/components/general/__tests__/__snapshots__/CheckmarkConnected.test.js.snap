// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckmarkConnected renders initial component 1`] = `
<span
  className="btn checkmark-btn"
  onClick={[Function]}
>
  <span
    className="checkmark-btn-label"
    style={
      Object {
        "paddingRight": "0px",
      }
    }
  >
    Husk
  </span>
  <div
    className=""
    style={
      Object {
        "display": "inline-block",
        "fontSize": "10px",
        "paddingLeft": "10px",
        "width": "20px",
      }
    }
  />
</span>
`;

exports[`CheckmarkConnected renders initial component with loggedIn user 1`] = `
<div
  className="checkmark-menu"
  onBlur={[Function]}
  tabIndex="1"
>
  <div>
    <span
      className="checkmark-menu--expand-btn"
    >
      <span
        className="btn btn-default"
        onClick={[Function]}
      >
        Husk
      </span>
      <span
        className="btn btn-default"
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-option-vertical"
        />
      </span>
    </span>
  </div>
  <div
    className="checkmark-menu--content list-group"
  />
</div>
`;
